deps/hiredis/sds.c

```diff
@@ -1020,7 +1020,13 @@ sds *sdssplitargs(const char *line, int *argc) {
                 if (*p) p++;
             }
             /* add the token to the vector */
-            vector = s_realloc(vector,((*argc)+1)*sizeof(char*));
+            char **data = s_realloc(vector,((*argc)+1)*sizeof(char*));
+            if (data == NULL) {
+            	return NULL;
```

Comment:
I guess the error handling should already be in s_alloc custom implementation; mainly freeing the previous memory blocks and then throwing an memory failure (with bailout from the process)